lib/internal/Magento/Framework/App/PageCache/Identifier.php

```diff
@@ -13,6 +13,23 @@
  */
 class Identifier implements IdentifierInterface
 {
+    /**
+     * Pattern detect marketing parameters
+     */
+    public const PATTERN_MARKETING_PARAMETERS = [
+        '/&?gclid\=[^&]+/',
+        '/&?cx\=[^&]+/',
+        '/&?ie\=[^&]+/',
+        '/&?cof\=[^&]+/',
+        '/&?siteurl\=[^&]+/',
+        '/&?zanpid\=[^&]+/',
+        '/&?origin\=[^&]+/',
+        '/&?fbclid\=[^&]+/',
+        '/&?mc_(.*?)\=[^&]+/',
+        '/&?utm_(.*?)\=[^&]+/',
+        '/&?_bta_(.*?)\=[^&]+/',
+    ];
+
```

Comment:
Probably will be good idea to use the same list as is in the varnish configuration: https://github.com/magento/magento2/blob/71432aebbc5211378cc3ebb4f90a6ad9ee08648e/app/code/Magento/PageCache/etc/varnish7.vcl#L84

Reply:
Yes, I used same list as varnish config